src/chainstate/coordinator/mod.rs

```diff
@@ -683,6 +1008,105 @@ impl<
+        let parent_block_id = self
+            .chain_state_db
+            .get_parent(&stacks_block_id)
+            .expect("BUG: failed to get parent for processed block");
+
+        // look up parent in exit_at_reward_cycle_info table
+        let exit_info_opt =
+            SortitionDB::get_exit_at_reward_cycle_info(self.sortition_db.conn(), &parent_block_id)?;
+
+        if let Some(parent_exit_info) = exit_info_opt {
+            info!("RCPR: cc: parent block info is non-None");
+            // copy info from parent block
+            current_exit_at_rc = parent_exit_info.curr_exit_at_reward_cycle;
+            if parent_exit_info.block_reward_cycle == current_reward_cycle {
+                current_proposal = parent_exit_info.curr_exit_proposal;
+            }
+            invalid_reward_cycles = parent_exit_info
+                .invalid_reward_cycles
+                .iter()
+                .filter(|rc| **rc > current_reward_cycle)
+                .map(|rc| *rc)
+                .collect();
+
+            // We process the previous reward cycle as soon as we transition to the next one.
+            if parent_exit_info.block_reward_cycle < current_reward_cycle {
+                info!("RCPR: cc: ancestor - rc is greater than parent rc");
+                // Check if there is some proposal. If so, need to check for a veto.
+                if let Some(curr_exit_proposal) = parent_exit_info.curr_exit_proposal {
+                    let veto_passed = self.read_veto_state(
+                        parent_exit_info.block_reward_cycle,
+                        curr_exit_proposal,
+                        &block_receipt.header.consensus_hash,
+                        &block_hash,
+                        exit_contract_id,
+                    )?;
+                    // if veto fails, record exit block height
+                    if !veto_passed {
+                        info!("RCPR: cc: veto did not pass for {}", curr_exit_proposal);
+                        current_exit_at_rc = Some(curr_exit_proposal);
+                    }
+                } else {
+                    // tally votes of previous reward cycle if there is no veto happening
+                    // if there is consensus for some proposal, record it
+                    current_proposal = self.tally_votes(
+                        parent_exit_info.block_reward_cycle,
+                        parent_exit_info.curr_exit_at_reward_cycle,
+                        parent_exit_info.invalid_reward_cycles,
+                        &block_hash,
+                        &block_receipt.header.consensus_hash,
+                        exit_contract_id,
+                    )?;
+                    if let Some(exit_proposal) = current_proposal {
+                        info!("RCPR: storing proposal info for {:?}", exit_proposal);
+                        // record the proposal in invalid_reward_cycles
+                        invalid_reward_cycles.push(exit_proposal);
```

Comment:
I think the name is throwing my understanding off.  This path is executed if the call to `.tally_votes()` indicated that since the parent block was mined, a proposed reward cycle has been voted upon (and may now be vetoed).  But why does that make the exit proposal "invalid"?  Does it just mean that we don't have to tally it again?

Reply:
It indicates that this particular reward cycle can no longer be voted for. For example, if the proposal to exit at this reward cycle is veto'd, it is not possible to propose this same exit reward cycle again.